src/abstractdataframe/abstractdataframe.jl

```diff
@@ -486,11 +486,16 @@ Get the first row of `df` as a `DataFrameRow`.
 Base.first(df::AbstractDataFrame) = df[1, :]
 
 """
-    first(df::AbstractDataFrame, n::Integer)
+    first(df::AbstractDataFrame, n::Integer, view::Bool=false)
 
 Get a data frame with the `n` first rows of `df`.
+
+If `view=false` a freshly allocated `DataFrame` is returned.
+If `view=true` then a `SubDataFrame` view into `df` is returned.
 """
-Base.first(df::AbstractDataFrame, n::Integer) = df[1:min(n, nrow(df)), :]
+function Base.first(df::AbstractDataFrame, n::Integer, view::Bool=false) 
```

Comment:
I think that `view` should be a kwarg. @nalimilan - can you please say what you prefer before we make the change?

Reply:
OK - so given the 👍 - please change it to kwarg.